src/core/amount.js

```diff
@@ -148,20 +156,22 @@ Amount.prototype.subtract = function(subtrahend) {
 
 // XXX Diverges from cpp.
 Amount.prototype.multiply = function(multiplicand) {
-  var multiplicandAmount = Amount.from_json(multiplicand);
-  // TODO: probably should just multiply by multiplicandAmount._value
-  var multiplyBy = multiplicandAmount.is_native() ?
-    multiplicandAmount._value.times(Amount.bi_xns_unit)
-    : multiplicandAmount._value;
-  return this._copy(this._value.times(multiplyBy));
+
+  const multiplicandAmount = Amount.from_json(multiplicand);
+  const multiplyBy = multiplicandAmount.is_native() ?
+    new Value_XRP(multiplicandAmount._value, Amount.bi_xns_unit)
+    : new Value_IOU(multiplicandAmount._value);
+  const thisValue = new Value_IOU(this._value);
+  return this._copy(new BigNumber(thisValue.multiply(multiplyBy)._value));
```

Comment:
I don't think we need an amount class. The plan is to factor out `XRPValue` and `IOUValue` into a ripple-lib-extensions module. Client code can do math with these value objects. The only thing the Amount class has besides math is some parsing and formatting, which can be removed from ripple-lib and into a separate ripple-lib-extensions module (only needed by UI code). Of course it also keeps the currency and counterparty, but it doesn't actually track them properly for multiply and divide, so this isn't really useful. Then we could potentially replace the Amount class with just a plain javascript object: `{currency, counterparty, value}`.